### ISSUE TITLE
Microsoft.Graph version issue

### ISSUE THREAD

**Issue by patrikhellgren** [pnp/pnpframework#660]
When using PnP.Framework 1.9 (wich references Microsoft.Graph version 3.33) in a project that itself references Microsoft.Graph (latest stable 4.28) there are some issues when calling PnP.Framework methods that uses Microsoft.Graph (e.g. all methods in the UnifiedGroupsUtility class). I get an exception saying "Method not found".

The issue goes away if either updating PnP.Framework to use Microsoft.Graph 4.28 or downgrading our project to use 3.33.

Is there a known way of having PnP.Framework not be that version dependant for Microsoft.Graph? If not, would it be ok to bump the Microsoft.Graph NuGet package (and it's dependencies) to the latest stable (3.33 is after all a year old)? I would gladly make a PR for that since I already have it finished as a test.

Environment:
PnP.Framework 1.9 (also tried with latest daily)
.Net 6
Tested both in an Azure Function v4 (Windows) and in LINQPad 7 running on Windows 11 with the same results.

**Comment by jansenbe** [comment]
Hey  @patrikhellgren : Adding @KoenZomers/ @gautamdsheth  / @erwinvanhunen here (Koen is one of the main authors of the PnP Framework code using Graph SDK) + for PowerShell impact. Overall this seems to be a simple bump, but we need to carefully review this as we'll want shared dependencies between PnP Core, PnP Framework and PnP PowerShell to stay aligned. Also from the PS side bumping certain dependencies might lead to assembly load conflicts when PS is used in Azure. So the exercise to do is a global "review and bump" dependencies for PnP Core / PnP Framework and PnP PowerShell

**Comment by patrikhellgren** [comment]
@jansenbe the mention of KoenZomers is not correct so adding him here again. @KoenZomers please check Bert's previous comment in this thread.

**Comment by KoenZomers** [comment]
I believe to remember we were having some weird constraints which made us be stuck on an old Graph version. Actually believe to have heard this from you in the past @jansenbe :) My personal preference would be to disconnect entirely from the Microsoft Graph SDK and just use MSAL for the authentication and make raw Graph calls using a HttpClient. I don't see what it adds to use the Graph SDK. From PnP PowerShell perspective, in version 2, we want to break free from a lot of these dependencies by no longer natively passing back framework entities but creating mappers and returning our own entities. This will allow much greater flexibility.

For now, I'd say @patrikhellgren give it a try and bump the Graph SDK version and see if things will even compile. If you don't mind running some more extensive testing on the Graph backed functionality in PnP Core to discover if all works well, I guess we can consider upgrading at least in a nightly build. This will allow us and a bit larger audience to test drive this in a bit larger scale before we publish a next release.

Would this be an okay approach to you as well @jansenbe?

**Comment by jansenbe** [comment]
@KoenZomers : I'm all for lowering our dependencies and if so moving away from the Graph SDK. MSAL based auth is already part of PnP Framework and there's a base REST client that can be used (think that's how we initially did this). This however will require a rewrite of all the code that depends on the SDK. Also PnP PS has a direct reference to this Graph SDK, so that would also be reviewed for usage and if needed updated.

I personally don't have the bandwidth to work on these rewrites, but if @KoenZomers / @patrikhellgren / @gautamdsheth or others have some time then I suggest doing the work to move PnP Framework and PnP PowerShell away from Graph SDK. PnP Core SDK is not using it. If the needed Graph usage in PnP Framework is Teams related, then using these features from PnP Core SDK might save some time. 

Once that's done we can evaluate bumping the auth dependencies for the whole stack (PnP Core, PnP Framework and PnP PowerShell)

**Comment by luismanez** [comment]
Is there any update here? we´re also facing this issue. Our project is using a pretty up to date version of the package ```Microsoft.Identity.Web.MicrosoftGraph```, which is referencing ```Microsoft.Graph 4.11```

We´ve done huge effort to move our product to .NET 6, Azure Functions v4 and latest packages, and is a bit painful to be now in this situation.

I also don´t understand why this happens, as in theory, PnP is not stuck to a specific version in Graph SDK (like it was in the past with Newtonsoft). Image below says it should support Microsoft.Graph 3.33 or greater:

![image](https://user-images.githubusercontent.com/9392197/176621465-89dfb2b8-6b73-4202-8a05-846805f8ab95.png)

Any update would be highly appreciated.

**Comment by martinlingstuyl** [comment]
Any updates here? We're also facing this issue. Just after rewriting our code to use the latest Microsoft Graph SDK... (5.11.0)

**Comment by martinlingstuyl** [comment]
It seems to me Microsoft Graph SDK is the new Newtonsoft. @KoenZomers any update? @luismanez what did you do to work around this?

**Comment by patrikhellgren** [comment]
I never got around to doing any extensive testing after bumping the dependency on Microsoft.Graph in PnP.Framework to anything later than 3.33 so we are still using that version in our solutions which of course is not optimal. We will probably also need the additions in the later versions shortly so great if this could get some attention again.

**Comment by martinlingstuyl** [comment]
> Also note that if you're using SDK 5.x, it has quite a few breaking changes from 4.x (you know this, as you have re-written your code), so is not enough with upgrading PnP SDK package, and you're gonna need to change code in the PnP project...

Yes, I had just rewritten them before trying it out and discovering the pnp.framework issue 😔 fortunately, it wasn't a big job. I saved the changes somewhere so I can reuse them as soon as possible.

**Comment by gautamdsheth** [comment]
Hey @martinlingstuyl , @patrikhellgren - have created a draft PR to bump the version ? 

https://github.com/pnp/pnpframework/pull/859

**Comment by martinlingstuyl** [comment]
That's helpful!

**Comment by gautamdsheth** [comment]
It is 95% complete, there's 1 method is needs to be refactored and after that, will test it out.

**Comment by andrewbackway** [comment]
Thanks for the information, any update?

**Comment by TomekPi** [comment]
@gautamdsheth , any update on this? Are you targeting the Graph SDK 5.x or 4.x?

**Comment by nsyng** [comment]
@patrikhellgren @gautamdsheth any update on this?

**Comment by raclettierer** [comment]
any news ?

**Comment by quails4Eva** [comment]
Based on the earlier discussion about removing the Graph SDK nuget dependency I created a pr that starts switching away from the nuget package. @jansenbe does what I've done so look ok? If so I can try to keep picking away at it as I have time. #1003

**Comment by LockTar** [comment]
@quails4Eva any idea when #1003 is finished?
We have a lot of problems because of the Microsoft Graph nuget package reference...

**Comment by quails4Eva** [comment]
Not sure as I'm doing it in my spare time. Hopefully this year.

I'm also happy for anyone else to pick it up and finish it off, or do a bit so there's less for me to do.

2 Oct 2024 10:58:46 Ralph Jansen ***@***.***>:

**Comment by anthonywhite** [comment]
Anyone know what the latest is on this? We are getting runtime errors trying to use `1.18` in a solution that also uses Graph `5.x`...

**Comment by marcus905** [comment]
Any news here?

**Comment by quails4Eva** [comment]
> 何なにかニュースは？

Fix was merged in may last year #1003 .
I assume release is blocked by  #1192 .

**Comment by Magnus-K-work** [comment]
As of the release of PnP.Framework 1.19.0, the reference to Microsoft.Graph is removed.

BR
Magnus

**Comment by Adam-it** [comment]
@Magnus-K-work Thanks for the update and spreading the good news 👍
You Rock 🤩 and +1 to what you wrote
